### PR TITLE
Fix name of like function for connector expressions

### PR DESCRIPTION
**Pull request by martint**
The function is misnamed. It should have been $like all along.
$like_pattern is the name of the internal function to convert
from a varchar to the compiled form of a pattern (i.e., LikeMatcher).

## Documentation

(x) No documentation is needed.

## Release notes

(x) Release notes entries required with the following suggested text:

```markdown
# SPI
* Rename `LIKE_PATTERN_FUNCTION_NAME` to `LIKE_FUNCTION_NAME` in `StandardFunctions`. ({issue}`13965`)
```

### CODE REVIEW

**Review by wendigo — APPROVED** [review]
lgtm. thanks!

**Comment by wendigo** [comment]
Mind the CI:
```
Error:  io.trino.operator.scalar.TestLikeFunctions.testFunctionNameConstantsInSync  Time elapsed: 0.01 s  <<< FAILURE!
java.lang.AssertionError: expected [$like_pattern] but found [$like]
	at org.testng.Assert.fail(Assert.java:94)
	at org.testng.Assert.failNotEquals(Assert.java:513)
	at org.testng.Assert.assertEqualsImpl(Assert.java:135)
	at org.testng.Assert.assertEquals(Assert.java:116)
	at org.testng.Assert.assertEquals(Assert.java:190)
	at org.testng.Assert.assertEquals(Assert.java:200)
	at io.trino.operator.scalar.TestLikeFunctions.testFunctionNameConstantsInSync(TestLikeFunctions.java:56)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
Would be nice to update `io.trino.plugin.jdbc.expression.TestExpressionMappingParser#testCallPattern` too

File: core/trino-spi/src/main/java/io/trino/spi/expression/StandardFunctions.java [hunk 1]
```diff
@@ -81,7 +81,7 @@ private StandardFunctions() {}
      */
     public static final FunctionName NEGATE_FUNCTION_NAME = new FunctionName("$negate");
 
-    public static final FunctionName LIKE_PATTERN_FUNCTION_NAME = new FunctionName("$like_pattern");
```
Comment (findepi) [inline core/trino-spi/src/main/java/io/trino/spi/expression/StandardFunctions.java:84]: The old name is what we use internally: `io.trino.type.LikeFunctions#LIKE_PATTERN_FUNCTION_NAME`

I think the internal name should be updated accordingly.

**Pull request merged**